File: 03-poo/07-sobrescritura-y-polimorfismo/java/ejemplos/Polimorfismo.java
public class Polimorfismo {
    static abstract class Notificacion { abstract void enviar(); }
    static class Email extends Notificacion { @Override void enviar(){ System.out.println("Email enviado"); } }
    static class Sms extends Notificacion { @Override void enviar(){ System.out.println("SMS enviado"); } }
    public static void main(String[] args) {
        Notificacion[] canales = { new Email(), new Sms() };
        // La misma llamada ejecuta el comportamiento del objeto real.
        for (Notificacion canal : canales) canal.enviar();
    }
}

File: 03-poo/07-sobrescritura-y-polimorfismo/scala/ejemplos/Polimorfismo.scala
trait Notificacion:
  def enviar(): Unit
class Email extends Notificacion:
  def enviar(): Unit = println("Email enviado")
class Sms extends Notificacion:
  def enviar(): Unit = println("SMS enviado")

@main def polimorfismo(): Unit =
  val canales: List[Notificacion] = List(new Email, new Sms)
  canales.foreach(_.enviar())
